public/js/highlight.js

```diff
@@ -1,2 +1,33 @@
\ No newline at end of file
+'use strict';
+
+/* globals window */
+
+var hljs = require('highlight.js/lib/highlight');
+
+window.hljs = hljs;
+
+hljs.registerLanguage('apache', require('highlight.js/lib/languages/apache'));
+hljs.registerLanguage('bash', require('highlight.js/lib/languages/bash'));
+hljs.registerLanguage('cs', require('highlight.js/lib/languages/cs'));
+hljs.registerLanguage('cpp', require('highlight.js/lib/languages/cpp'));
+hljs.registerLanguage('css', require('highlight.js/lib/languages/css'));
+hljs.registerLanguage('coffeescript', require('highlight.js/lib/languages/coffeescript'));
+hljs.registerLanguage('diff', require('highlight.js/lib/languages/diff'));
+hljs.registerLanguage('xml', require('highlight.js/lib/languages/xml'));
+hljs.registerLanguage('http', require('highlight.js/lib/languages/http'));
+hljs.registerLanguage('ini', require('highlight.js/lib/languages/ini'));
+hljs.registerLanguage('json', require('highlight.js/lib/languages/json'));
+hljs.registerLanguage('java', require('highlight.js/lib/languages/java'));
+hljs.registerLanguage('javascript', require('highlight.js/lib/languages/javascript'));
+hljs.registerLanguage('makefile', require('highlight.js/lib/languages/makefile'));
+hljs.registerLanguage('markdown', require('highlight.js/lib/languages/markdown'));
+hljs.registerLanguage('nginx', require('highlight.js/lib/languages/nginx'));
+hljs.registerLanguage('objectivec', require('highlight.js/lib/languages/objectivec'));
+hljs.registerLanguage('php', require('highlight.js/lib/languages/php'));
+hljs.registerLanguage('perl', require('highlight.js/lib/languages/perl'));
+hljs.registerLanguage('python', require('highlight.js/lib/languages/python'));
+hljs.registerLanguage('ruby', require('highlight.js/lib/languages/ruby'));
+hljs.registerLanguage('sql', require('highlight.js/lib/languages/sql'));
+hljs.registerLanguage('shell', require('highlight.js/lib/languages/shell'));
```

Comment:
Do you think perhaps we should allow this list to be configurable? Perhaps scan `lib/languages` and output the filenames in a multi-select?

Reply:
Additionally I suppose if new languages get added then no work will be required to add support in the plugin.

Reply:
https://github.com/highlightjs/highlight.js/tree/master/src/languages Wow there are a lot of languages :sweat_smile: 

Might be good to allow configuration for sure.

Reply:
Yes, there are a lot of languages. If you include all of them it will more than double the current `nodebb.min.js` file. Which seems a bit questionable? This gets even more questionable if you build this and include it in the `nodebb.min.js` but then don't use it. That is more than 50% of dead code the users have to download and parse. You increase load time for literally nothing.

So if you want to do this, in my opinion it has to happen in a way that those languages won't get included in `nodebb.min.js`. Best would be if every language gets its own file that can be loaded at runtime. And I have to say I'm currently not entirely sure how to do this properly with the plugin loading system of NodeBB. I mainly have two problems:

* How to configure the `plugin.json` properly for this use case
* How to automatically include new languages in the `plugin.json` file. I'd rather not have to configure this by hand every time new languages get supported.

To approach this proposal I would at least suggest a different approach. Currently this pull request recreates what is currently configured in the plugin. At the moment not all languages are supported. If someone wants to support more languages, they can fork from this plugin and add them to the `highlight.js`. This will be much easier as it was before this pull request. And then lets open a new issue to look into properly supporting additional languages via configuration.

Reply:
Just FYI webpack supports code splitting and will create multiple bundles of you set it up correctly.

Usually only requires an `import(...)` call and everything else is handled for you. My calendar plugin could work as an example.

Reply:
Yes, I'm aware of this. But currently it is outputting AMD modules and thus NodeBB ist putting everything back together into `nodebb.min.js`.